packages/components/input-tag/src/input-tag.vue

```diff
@@ -10,25 +10,108 @@
+        </el-tag>
+        <el-tag
+          v-if="!collapseTags"
+          :size="tagSize"
+          :closable="closable"
+          :type="tagType"
+          :effect="tagEffect"
+          :draggable="closable && draggable"
+          disable-transitions
+          @close="handleRemoveTag(index)"
+          @dragstart="(event: DragEvent) => handleDragStart(event, index)"
+          @dragover="(event: DragEvent) => handleDragOver(event, index)"
+          @dragend="handleDragEnd"
+          @drop.stop
+        >
+          <slot name="tag" :value="item" :index="index">
+            {{ item }}
+          </slot>
+        </el-tag>
+      </template>
       <el-tag
-        v-for="(item, index) in modelValue"
-        :key="index"
+        v-if="
+          collapseTags &&
+          !collapseTagsTooltip &&
+          modelValue &&
+          modelValue.length > maxCollapseTags
+        "
+        :closable="false"
         :size="tagSize"
-        :closable="closable"
         :type="tagType"
         :effect="tagEffect"
-        :draggable="closable && draggable"
         disable-transitions
-        @close="handleRemoveTag(index)"
-        @dragstart="(event: DragEvent) => handleDragStart(event, index)"
-        @dragover="(event: DragEvent) => handleDragOver(event, index)"
-        @dragend="handleDragEnd"
-        @drop.stop
       >
-        <slot name="tag" :value="item" :index="index">
-          {{ item }}
-        </slot>
+        + {{ modelValue.length - maxCollapseTags }}
       </el-tag>
+      <el-tooltip
+        v-if="
+          collapseTags &&
+          collapseTagsTooltip &&
+          modelValue &&
+          modelValue.length > maxCollapseTags
+        "
+        ref="tagTooltipRef"
+        :fallback-placements="['bottom', 'top', 'right', 'left']"
+        :effect="tagEffect"
+        placement="bottom"
+      >
+        <template #default>
+          <el-tag
+            :closable="false"
+            :size="tagSize"
+            :type="tagType"
+            :effect="tagEffect"
+            disable-transitions
+          >
+            + {{ modelValue.length - maxCollapseTags }}
+          </el-tag>
+        </template>
+        <template #content>
+          <div :class="ns.e('input-tag-list')">
+            <div
+              v-for="(item, index) in modelValue"
+              :key="item"
+              :class="ns.e('input-tag-item')"
+            >
+              <el-tag
+                v-if="index >= maxCollapseTags"
+                :size="tagSize"
+                :closable="closable"
+                :type="tagType"
+                :effect="tagEffect"
+                disable-transitions
+                @close="handleRemoveTag(index)"
+              >
+                <span :class="ns.e('tags-text')">
+                  <slot name="label" :value="item" :index="index">
+                    {{ item }}
+                  </slot>
+                </span>
+              </el-tag>
+            </div>
```

Comment:
IMO, It would be better to calculate collapseTagList like ElSelect components

It seems that unnecessary nested node `input-tag-item`, `tags-text` can be avoided here